Source/WebCore/rendering/EventRegion.cpp

```diff
@@ -208,8 +208,44 @@ bool EventRegionContext::shouldConsolidateInteractionRegion(IntRect bounds, Rend
     return false;
 }
 
+void EventRegionContext::addOcclusionMarginsForSmallInteractionRegions()
+{
+    static int minimumSize = 20;
+    static int occlusionMargin = 10;
+    Vector<InteractionRegion> interactionRegionsWithOcclusions;
+    bool addedOcclutions = false;
+    for (auto& interactionRegion : m_interactionRegions) {
+        bool isSmallRect = false;
+        auto occlusionRect = interactionRegion.rectInLayerCoordinates;
+
+        if (interactionRegion.rectInLayerCoordinates.width() < minimumSize) {
+            occlusionRect.inflateX(occlusionMargin);
+            isSmallRect = true;
+        }
+        
+        if (interactionRegion.rectInLayerCoordinates.height() < minimumSize) {
+            occlusionRect.inflateY(occlusionMargin);
+            isSmallRect = true;
+        }
+            
+        if (isSmallRect) {
+            interactionRegionsWithOcclusions.append({
+                interactionRegion.elementIdentifier,
+                occlusionRect,
+                interactionRegion.borderRadius,
+                InteractionRegion::Type::Occlusion
+            });
+            addedOcclutions = true;
+            interactionRegionsWithOcclusions.append(interactionRegion);
```

Comment:
I don't have full context into this change, but is it intentional that the original `interactionRegion` is only added if `isSmallRect` is true?

In the case where `m_interactionRegions` contains two regions, and one is deemed "small" and the other is not, the latter region would get dropped from `m_interactionRegions`. Is that scenario possible and is the described behavior desirable?